cupyx/scipy/interpolate/_polyint.py

```diff
@@ -0,0 +1,332 @@
+import cupy
+from cupyx.scipy._lib._util import _asarray_validated, float_factorial
+
+
+def _isscalar(x):
+    """Check whether x is if a scalar type, or 0-dim"""
+    return cupy.isscalar(x) or hasattr(x, 'shape') and x.shape == ()
+
+
+class _Interpolator1D:
+    """Common features in univariate interpolation.
+
+    Deal with input data type and interpolation axis rolling. The
+    actual interpolator can assume the y-data is of shape (n, r) where
+    `n` is the number of x-points, and `r` the number of variables,
+    and use self.dtype as the y-data type.
+
+    Attributes
+    ----------
+    _y_axis : Axis along which the interpolation goes in the
+        original array
+    _y_extra_shape : Additional shape of the input arrays, excluding
+        the interpolation axis
+    dtype : Dtype of the y-data arrays. It can be set via _set_dtype,
+        which forces it to be float or complex
+
+    Methods
+    -------
+    __call__
+    _prepare_x
+    _finish_y
+    _reshape_y
+    _reshape_yi
+    _set_yi
+    _set_dtype
+    _evaluate
+
+    """
+
+    def __init__(self, xi=None, yi=None, axis=None):
+        self._y_axis = axis
+        self._y_extra_shape = None
+        self.dtype = None
+        if yi is not None:
+            self._set_yi(yi, xi=xi, axis=axis)
+
+    def __call__(self, x):
+        """Evaluate the interpolant
+
+        Parametres
```

Comment:
Maybe building doc fails because of this.
```suggestion
        Parameters
```

Reply:
Oops 😅